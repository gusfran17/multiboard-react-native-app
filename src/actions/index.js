export {
    ADD_PLAYER,
    REMOVE_PLAYER,
    UPDATE_SCORE,
    UPDATE_PLAYER_STATUS,
    SELECT_PLAYER,
    UPDATE_DISPLAY_STATS,
    UPDATE_GAME_STATUS,
    START_NEW_GAME,
    SAVE_PROGRESS,
    LOAD_GAME,
    addPlayer,
    removePlayer,
    updateScore,
    updatePlayerStatus,
    selectPlayer,
    UPDATE_WIN_OR_LOSE,
    UPDATE_MAX_SCORE,
    updateWinOrLose,
    updateMaxScore,
    updateDisplayStats,
    updateGameStatus,
    startNewGame,
    saveProgress,
    loadGame
} from './game';
export {
    UPDATE_NEW_GAME_WIN_OR_LOSE,
    UPDATE_NEW_GAME_MAX_SCORE,
    SAVE_GAME,
    UPDATE_GAME,
    REMOVE_GAME,
    updateNewGameWinOrLose,
    updateNewGameMaxScore,
    saveGame,
    updateGame,
    removeGame
} from './mainMenu';
