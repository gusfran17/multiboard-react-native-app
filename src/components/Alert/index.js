export Alert from './Alert';
export CustomAlert from './CustomAlert';
