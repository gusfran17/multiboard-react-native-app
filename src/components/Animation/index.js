export BringFromBottom from './BringFromBottom';
export GrowToHeight from './GrowToHeight';
