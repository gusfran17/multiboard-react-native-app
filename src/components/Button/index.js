export StatefullAnimatedButton from './StatefullAnimatedButton';
export AnimatedButton from './AnimatedButton';
