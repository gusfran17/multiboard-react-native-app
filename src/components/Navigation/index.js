export NavigationHeader from './NavigationHeader';
export RightHeader from './RightHeader';
export LeftHeader from './LeftHeader';
