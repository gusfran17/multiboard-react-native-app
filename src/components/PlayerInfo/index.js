export PlayerInfo from './PlayerInfo';
export Detail from './Detail';
