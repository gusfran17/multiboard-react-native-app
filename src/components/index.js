export {
    Settings
} from './Settings';
export {
    Scoreboard
} from './Scoreboard';
export {
    MainMenu
} from './MainMenu';
export {
    NavigationBar
} from './Navigation';
export {
    SavedGames
} from './SavedGames';
