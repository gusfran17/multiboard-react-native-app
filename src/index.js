import React from 'react';
import { Stack, } from './containers';
import Stopwatch from './components/Scoreboard/Stopwatch';

const Game = props => (
    <Stack/>
);

export default Game;
