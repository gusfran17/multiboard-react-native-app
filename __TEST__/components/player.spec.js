import { shallow, } from 'enzyme';
import * as React from 'react';
import { Player, Counter, } from './../../src/components/';

describe('<Player />', () => {
    it('should render as expected', () => {
        const wrapper = shallow(<Player
            name="Gustavo"
            index={0}
            updateScore={() => {}}
            removePlayer={() => {}}
            addPlayer={() => {}}
            score={20}
        />);
        expect(wrapper).toMatchSnapshot();
    });

    it('should have a Counter', () => {
        const wrapper = shallow(<Player
            name="Gustavo"
            index={0}
            updateScore={() => {}}
            removePlayer={() => {}}
            addPlayer={() => {}}
            score={20}
        />);
        expect(wrapper.find('Counter').length).toBe(1);
    });
});
