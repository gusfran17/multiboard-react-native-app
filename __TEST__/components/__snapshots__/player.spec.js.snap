// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Player /> should render as expected 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#2f2f2f",
      "borderBottomColor": "#444",
      "borderBottomWidth": 2,
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      Object {
        "flex": 4,
        "flexDirection": "row",
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#900",
            "fontSize": 15,
            "fontWeight": "900",
            "paddingBottom": 19,
            "paddingLeft": 10,
            "paddingRight": 0,
            "paddingTop": 19,
          }
        }
      >
        âœ–
      </Text>
    </TouchableOpacity>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#fff",
          "flex": 1,
          "fontWeight": "900",
          "marginLeft": -30,
          "paddingBottom": 20,
          "paddingLeft": 5,
          "paddingRight": 5,
          "paddingTop": 20,
          "textAlign": "center",
        }
      }
    >
      GUSTAVO
    </Text>
  </View>
  <Counter
    index={0}
    score={20}
    updateScore={[Function]}
  />
</View>
`;
